app/controllers/api/v1/admin/users_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Api::V1::Admin::UsersController < Api::V1::BaseController
+  before_action :authenticate_user!
+  before_action :authorize_user
+  before_action :load_user, except: [:index, :roles, :statuses]
+  after_action :verify_authorized
+
+  def index
+    @users = User.all
+    respond_with @users
+  end
+
+  def roles
+    respond_with User.roles.keys
+  end
+
+  def statuses
+    respond_with User.statuses.keys
+  end
+
+  def salary
+    respond_with @user.salaries.last.amount
+  end
+
+  def destroy
+    respond_with @user.destroy
+  end
+
+  def set_salary
+    if ((@user.salaries.last.update_attributes(archived_at: params[:archived_at])) && (@user.salaries.create(amount: params[:amount])))
+      respond_with @user, json: @user
+    end
+  end
+
+  def update
```

Comment:
this is a good example why should salary go to completely different controller. We have `Users` controller and `update` action. The logic says that we should have `user` update here.